source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -981,6 +1007,26 @@ See the `Load Balancer Specification <../load-balancers/load-balancers.rst#event
        *  The ID of the Connection
        */
       connectionId: int64;
+
+      /**
+       * The time elapsed waiting for conditions specified by `ConnectionPoolOptions.waitQueueTimeoutMS`.
+       * More specifically, the time elapsed between the `ConnectionCheckOutStartedEvent` and:
+       *  - the `ConnectionCreatedEvent` emitted by the same checking out, if this checking out emitted `ConnectionCreatedEvent`;
+       *  - this event, otherwise.
```

Comment:
> As a result, the average check-out duration actually goes down

The important thing to note here is that `ConnectionCheckedOutEvent`/`ConnectionCheckOutFailedEvent.waited` is not a "check-out duration", and it is not documented as such.

> Instead, we should always report the entire time between when a check-out starts and when it delivers a connection to the caller.

The description of https://jira.mongodb.org/browse/DRIVERS-2469 requests the waited time and time to establish, which is what I added.

If the checkout time appears to be also needed, we may

1. either replace the waited time with the checkout time,
2. or add the checkout time to be the third reported duration.

Is my understanding correct that you prefer the first option?

Reply:
here's the phrasing from the ticket:
> - The amount of time a request to check out a connection waited for an idle or new connection.
> - The total amount of time a new connection took to establish (i.e. the time between requesting a new connection and the new connection being ready to send application operations on).

I agree with Matt here, for the reasons that Matt described.  But also because I interpret the request in the drivers ticket to deliver the following new metrics:

- the whole time it took for any checkout to succeed or fail (the time between `ConnectionCheckoutStarted` and `ConnectionCheckoutSucceeded` or `ConnectionCheckoutFailed`)
- the time it takes for any connection to establish (the time between `ConnectionCreated` and `ConnectionReady`)

As it currently stands, a user would not be able to interpret the `duration` property on the `ConnectionCheckoutSucceededEvent` without monitoring for connection establishment and correlating the events.  I'd argue that this makes the `duration` properties meaningless as its currently specified.

If we make the `duration` on the `ConnectionCheckoutSucceededEvent` the time between `ConnectionCheckoutStarted` and the successful connection checkout, users might see spikes in this value when new connections are created and they can confirm this corresponds to new connections being established if they wish, but the duration can be interpreted by itself without monitoring for other events.

Reply:
That's correct, I think we should replace "waited" time with the full checkout duration. My intent in DRIVERS-2469 was not to distinguish separate time waited for an idle connection and time waited for a new connection, but to add two mostly orthogonal metrics (check-out duration and connection establishment duration).

Reply:
Done.